src/sentry/workflow_engine/endpoints/organization_data_condition_index.py

```diff
@@ -0,0 +1,75 @@
+from drf_spectacular.utils import extend_schema
+from rest_framework import serializers
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import OrganizationEndpoint
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.apidocs.constants import (
+    RESPONSE_BAD_REQUEST,
+    RESPONSE_FORBIDDEN,
+    RESPONSE_NOT_FOUND,
+    RESPONSE_UNAUTHORIZED,
+)
+from sentry.apidocs.parameters import GlobalParams
+from sentry.apidocs.utils import inline_sentry_response_serializer
+from sentry.workflow_engine.endpoints.serializers import (
+    DataConditionHandlerResponse,
+    DataConditionHandlerSerializer,
+)
+from sentry.workflow_engine.models.data_condition import LEGACY_CONDITIONS
+from sentry.workflow_engine.registry import condition_handler_registry
+from sentry.workflow_engine.types import DataConditionHandler
+
+
+@region_silo_endpoint
+class OrganizationDataConditionIndexEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.ISSUES
+
+    @extend_schema(
+        operation_id="Fetch Data Conditions",
+        parameters=[
+            GlobalParams.ORG_ID_OR_SLUG,
+        ],
+        responses={
+            201: inline_sentry_response_serializer(
+                "ListDataConditionHandlerResponse", list[DataConditionHandlerResponse]
+            ),
+            400: RESPONSE_BAD_REQUEST,
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def get(self, request, organization):
+        """
+        Returns a list of data conditions for a given org
+        """
+        group = request.GET.get("group")
+        try:
+            DataConditionHandler.Group(group)
+        except ValueError:
+            raise serializers.ValidationError("Invalid group")
```

Comment:
I'd expect a more descriptive error if `?group` isn't passed at all. It's also really helpful to list the valid options when you're throwing a validation error like this